### ISSUE TITLE
Example of SigMF Interpreter

### ISSUE THREAD

**Issue by cadeallen40** [TorchDSP/torchsig#257]
Hello,

I'm struggling to use torchsig.datasets.file_datasets.SigMFInterpreter to create a dataset. I'd like to load individual signals in SigMF format to create a dataset made up of those signals, but I'm not sure if this is possible with the SigMFInterpreter. How do we use the SigMFInterpreter to create datasets?

Thanks!

**Comment by MattCarrickPL** [comment]
Thanks for bringing this to our attention. We've developed a new example notebook, number 12, that describes how to load binary files through SigMF. We also reworked some of the code we use to interface with SigMF since it seems to have gotten out of sync with the rest of the codebase. We are expecting to release the notebook as part of the next 0.6.1 release.

One note of warning:

TorchSig 0.6.x (and earlier) uses an implied, normalized sampling rate of fs=1, therefore all of the fields within the SigMF metadata also need to conform to this sampling rate. This is not ideal, as it requires another set of conversion from actual RF frequencies to TorchSig's fs=1. However, a future release will allow TorchSig to use a parameterizable sampling rate which will make using SigMF recordings much easier down the line.

I'm going to leave this open for feedback until after the 0.6.1 release, so please send any questions you have after reviewing the new notebook.

**Comment by cadeallen40** [comment]
Sounds good, I'll take a look when the next release happens!

Would it be possible to use SigMF data to create image datasets? Say if I wanted to load examples of different signals as multiple SigMF files, then apply a SpectrogramImage transform to that like example 9 with the narrowband data. 

My end goal is to create a hybrid dataset that uses individual signals recorded in SigMF format that are then used to randomly create synthetic spectrogram images w/labels for training YOLO.

**Comment by MattCarrickPL** [comment]
Yes, you should be able to do this. After you try this in 0.6.1 and still have questions please let us know.

**Comment by cadeallen40** [comment]
Hello, I really appreciate the effort you have put into this project. I was wondering if there's an update on when the 0.6.1 release might be available? I'm particularly interested in the SigMF functionality.

**Comment by MattCarrickPL** [comment]
I don't have a specific date other than to say it is close. Maybe that is a week, maybe that is a month, I can't be more specific. Last week the team and I were going through the example notebooks to make sure they are all working correctly after some of the code changes we made, which is one of the last checks before a release goes out. As part of this release we are planning to upload better pre-trained models and we're currently in the process of verifying and validating them for performance before we release them to the public.

**Comment by ereoh** [comment]
@cadeallen40 with the release of v0.6.1, take a look at `examples/12_example_sigmf.ipynb`. I'll close this issue for now, but feel free to reopen it as needed.
